src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -124,12 +124,26 @@ export default class AuthCodeRedirectHandler implements IRedirectHandler {
       await this.fetcher.fetch(issuerConfig.tokenEndpoint, tokenRequestInit)
     ).json();
 
-    // TODO: should handle error if the token response is something strange
+    // Check the response
+    if (
+      !(
+        tokenResponse &&
+        tokenResponse.access_token &&
+        tokenResponse.id_token &&
+        typeof tokenResponse.access_token === "string" &&
+        typeof tokenResponse.id_token === "string" &&
+        (!tokenResponse.refresh_token ||
+          typeof tokenResponse.refresh_token === "string")
+      )
+    ) {
+      throw new Error("IDP /token route returned an invalid response.");
```

Comment:
Is the token endpoint always going to be `/token`, or should the error message use `issuerConfig.tokenEndpoint` instead ?

Reply:
Good catch

Reply:
And the value itself should be delimited in the message string (I always use '[' and ']'), as this helps see if the value has leading or trailing spaces.